SCAutolib/models/file.py

```diff
@@ -0,0 +1,104 @@
+on config files:
+    create: create content of config file usually based on template.
+            Note, that some child classes may also update content of config
+            file if it already existed.
+    set:    modify values of config files, add keys or sections
+            if necessary
+    save:   save modified content to config file
+    clean:  remove config file; note that some child classes may also restore
+            original config file if backup exists.
+"""
+from configparser import ConfigParser
+from pathlib import Path
+from typing import Union
+
+from SCAutolib import logger
+
+
+class File:
+    """
+    This class defines an interface for generic operations on config files
+
+    create: create content of config file based on template file
+    set:    modify content of config files, add keys or sections if necessary
+    save:   save config file
+    clean:  remove config file
+    """
+    _conf_file = None
+    _template = None
+    _default_parser = None
+
+    def __init__(self, filepath: Union[str, Path], template: str = None):
+        """
+        Init of File
+
+        :param filepath: Path of config file
+        :type filepath: str of pathlib.Path
+        :param template: Path of template file
+        :type template: str
+        """
+        self._conf_file = Path(filepath)
+        if template is not None:
+            self._template = Path(template)
+
+    def create(self):
+        """
+        Populate internal parser object with content based on template.
+        """
+        if self._conf_file.exists():
+            logger.warning(f"Create error: {self._conf_file} already exists.")
+            raise FileExistsError(f'{self._conf_file} already exists')
+        else:
+            self._default_parser = ConfigParser()
+            self._default_parser.optionxform = str
+            with self._template.open() as t:
+                self._default_parser.read_file(t)
+
+    def set(self, key: str, value: Union[int, str, bool], section: str = None):
+        """
+        Modify value in config file.
+
+        :param key: value for this key will be updated
+        :type key: str
+        :param value: new value to be stored in [section][key] of config file
+        :type value: int or str or bool
+        :param section: section of config file that will be modified
+        :type section: str
+        """
+        if self._default_parser is None:
+            self._default_parser = ConfigParser()
+            self._default_parser.optionxform = str
+            with self._conf_file.open() as config:
+                self._default_parser.read_file(config)
+        if not self._default_parser.has_section(section):
+            logger.warning(f"Section {section} not present.")
+            logger.info(f"Adding section {section}.")
+            self._default_parser.add_section(section)
+
+        previous = self._default_parser.get(section, key, fallback="Not set")
```

Comment:
Does this work correctly if section is None?

Reply:
It tracebacks as I did not intend to support section-less config files here. I'm open to discuss and possibly re-evaluate desired behavior.

Reply:
We should be able to handle that pretty easily. Basically check if user provided section, if yes, check if that section exists and if not, add it.
And if the user didn't provide a section just append to the end of the file the user provided key,value

Reply:
Let's support only configparser compatible files - i.e. files with sections in this (Parent) class. User should provide section here. Child classes can modify methods to work differently.

Reply:
I feel it should be the other way around. This is a generic File class, we should not enforce the section parameter here.
But we should do it for the other child classes where we know the config file uses sections.

Reply:
I see your point. However, current implementation use configparser in file class. That means it will anyway tracebacks if section is not defined. I wrote it this way as we do not have general method that implements modifications of configs without sections. I mentioned it also in File class docstring. I see two options:

1) keeping File class in current design - only for config-parser compatible files with sections and than the check that section was specified should be here.
2) rewriting File class to support also section-less configs. That means it should accept parameter saying on what file it operates and based on that parameter set method should behave differently depending on presence/non-presence of sections in the config file.

Or did I miss something?

Reply:
You don't have to rewrite the class, you just have to change the set method to allow non-section edits.
Basically it should be something like `if user provided section, great, use configparser, if not open the file for appending and add f"{key} = {value}`